portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,28 @@ async function getMessageFromServer() {
   const quote = await response.text();
   document.getElementById('message-container').innerHTML = quote;
 }
+
+function login(url){
+  window.location = url;
+}
+
+async function setButtonName(){
+  const response = await fetch('/login');
+  const status = await response.text();
+  let obj = JSON.parse(status);
+  let element = document.getElementById("login-button");
+  if(!(obj.status.localeCompare("Login"))){
+      element.value = "Login";
+  }
+  else {
+      console.log(obj.status);
+      element.value = "Logout";
+  }
+}
+
+async function logInUser(){
```

Comment:
So I think what's happening now is this /login path is being called 2x unnecessarily.

On load, you make a request to /login to get the login button text ("Login" or "Logout") which also returns the login/logout URL.  But in the onload function (setButtonName), all you do is set the text of the button.  If the user clicks the button, you make an additional request to /login to get the URL and then redirect to that URL.  But you already have the URL since you got it onload! 

Perhaps you can make this element an anchor tag and then update the innerText and the href values when the setButtonName comes back with the data from the server.  Then you won't need any click handlers on the element, since it will just be a link with the URL you set.